docs/RFCS/enterprise.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: Enterprise Registration
+- Status: draft
+- Start Date: 2017-01-17
+- Authors: Daniel Harrison and David Taylor
+- RFC PR: (PR # after acceptance of initial draft)
+
+# Summary
+
+CockroachDB's [business model][] describes a set of "enterprise features", which
+(after an evaluation period) must be purchased to be used by commercial
+entities. This document describes the technical mechanisms involved.
+
+# Detailed design
+
+An enterprise license is represented as a protobuf. It contains the user
+organization's name, a valid_until date (for an N-day evaluation license), a
+cluster_id, and whatever other information is necessary. The protobuf is
+serialized and base64 encoded.
+
+A new system table, `licenses`, contains each license and the timestamp of when
+it was added. It is located in the system config keyspace and so is gossiped to
+every node. When a SQL statement using an enterprise feature is executed, the
```

Comment:
This needs to be cached. 
A read-only cache with no expiration is sufficient as we can probably assume licenses are not modified in-place.